lib/node/api.js

```diff
@@ -965,3 +957,117 @@ api.getFunds = async (req, res) => {
+        totalInputValue += value;
+        if (address === scriptPubKey.addresses[0]) {
+          txDetail.transactionType = 'outgoing';
+        }
+      }
+      totalOutputValue = txDetail.vout.reduce((acc, { value }) => acc + value, 0);
+      const fee = totalInputValue - totalOutputValue;
+      txDetail.totalInputValue = totalInputValue.toFixed(8);
+      txDetail.totalOutputValue = totalOutputValue.toFixed(8);
+      txDetail.fee = fee.toFixed(8);
+      allTxDetails.push(txDetail);
+    }
+    const { result: unspents } = await btcClient('getaddressutxos', [{ addresses: [address] }], isOmni);
+    const confAmount = unspents.reduce((prev, curr) => prev + curr.satoshis, 0);
+    const unconfAmount = mempoolTx.reduce((prev, curr) => prev + curr.satoshis, 0);
+    return res.status(200).send({
+      error: null,
+      message: 'Transactions obtained successfully.',
+      result: {
+        txns: allTxDetails,
+        balance: {
+          confirmed: Number((confAmount / 10 ** 8).toFixed(8)),
+          unconfirmed: Number((unconfAmount / 10 ** 8).toFixed(8)),
+          total: Number(((confAmount + unconfAmount) / 10 ** 8).toFixed(8)),
+        },
+      },
+    });
+  } catch (errT) {
+    if (errT && errT.message) {
+      return res.status(500).send({ error: errT.message, result: null });
+    }
+    return res.status(500).send({ error: 'Internal error.', result: null });
+  }
+};
+
+/**
+ * API call to check if mempool allowed signed transaction to be inserted
+ */
+api.testMempoolAccept = async (req, res) => {
+  const { signedTx } = req.query;
+  let error = null;
+  try {
+    const { result } = (await btcClient('testmempoolaccept', [[signedTx]]));
+    if (result[0].allowed) {
+      return res.status(200).send({
+        message: 'mempool allowed this tx to be inserted',
+        result: result[0],
+        error,
+      });
+    }
+    error = result[0]['reject-reason'];
+  } catch (errT) {
+    if (errT && errT.message) {
+      error = errT.message;
+    } else {
+      error = 'Internal error.';
+    }
+  }
+  return res.status(500).send({ error, result: null });
+};
+
+/**
+ * API to obtain IPNS URL using transaction id
+ * @returns {object}
+ */
+api.getTxidIPNS = async (req, res) => {
+  const { txid } = req.query;
+  if (!txid) {
+    return res.status(400).send({ error: 'Send proper parameters.', result: null });
+  }
+  try {
+    const getRawTx = (await btcClient('getrawtransaction', [txid, true])).result;
+    const data = helpers.decodeNullDataIPNS(getRawTx.vout);
+    const message = data ? 'IPNS url fetched successfully' : 'No IPNS link found';
+    return res.status(200).send({
+      error: null,
+      message,
+      result: data,
+    });
+  } catch (errT) {
+    if (errT && errT.message) {
+      return res.status(500).send({ error: errT.message, result: null });
+    }
+    return res.status(500).send({ error: 'Internal error.', result: null });
```

Comment:
@mandeeps708 agreed. 'Internal error.' here is basically serving as a fallback error message if the message is not present. In any case the FE can just grab the error message and show it, also coz its hard to predict what the exact error code should be since its the parent try catch block.

@vikas-ranadev please in that same ticket please add this sub-task as well